### PR TITLE
Client message tracker service

### PR DESCRIPTION
**Pull request by albinsuresh**
Client message tracker service's interfaces for initial review. The relevant ones are:
1. ClientMessageTracker
2. MessageTracker
3. TrackerPolicy

The test package has some dummy entities generated to showcase the interaction between entities and this service.

### CODE REVIEW

**Comment by albinsuresh** [comment]
test this please

**Comment by albinsuresh** [comment]
test this please

**Comment by ljacomet** [comment]
Given the pattern of usage in the sample entity, and after discussing with @albinsuresh, why do we not have the following as the `MessageTracker` API:

```java
public interface MessageTracker<M extends EntityMessage, R extends EntityResponse> extends StateDumpable {
  R processMessage(InvokeContext context, M message, Function<M, R> processor);
}
```

**Comment by albinsuresh** [comment]
The new abstraction built over client message tracker is called "Once and only once message handler". If anyone has a better name to suggest please do so. Else I'll go ahead and modify the poms, packages etc as well to reflect the same name.

File: client-message-tracker/src/main/java/org/terracotta/client/message/tracker/OOOPolicy.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+package org.terracotta.client.message.tracker;
+
+import org.terracotta.entity.EntityMessage;
+
+/**
+ * A {@link OOOPolicy} for an entity defines which one of its message types need to be tracked by a {@link ClientMessageTracker}
+ */
+public interface OOOPolicy {
```
Comment (ljacomet) [inline client-message-tracker/src/main/java/org/terracotta/client/message/tracker/OOOPolicy.java]: Should be name with a more general name, such as `TrackingPolicy`

File: management/testing/integration-tests/pom.xml [hunk 2]
```diff
@@ -67,13 +67,6 @@
       <artifactId>galvan-support</artifactId>
       <version>${terracotta-core.version}</version>
       <scope>test</scope>
-      <!-- TODO Remove once galvan version is updated - simply check mockito-all is not longer pulled -->
```
Comment (ljacomet) [inline management/testing/integration-tests/pom.xml]: This cannot yet go, needs a later `terracotta-core` version

**Comment by ljacomet** [comment]
Just need one more rebase

**Pull request merged**